File: examples/node/echo.js
var faye = require('../../build/faye-node'),
    http = require('http');

var port = process.argv[2] || 7000;

var server = http.createServer();
server.addListener('upgrade', function(request, socket, head) {
  var socket = new Faye.WebSocket(request, head);
  socket.onmessage = function(message) {
    socket.send(message.data);
  };
});

server.listen(port);

File: examples/ruby/echo.rb
require 'rubygems'
require File.expand_path('../../../lib/faye', __FILE__)
require 'rack'
require 'thin'

port = ARGV[0] || 7000

app = lambda do |env|
  if env['HTTP_UPGRADE']
    socket = Faye::WebSocket.new(env)
    
    socket.onmessage = lambda do |event|
      socket.send(event.data)
    end
    
    socket.onclose = lambda do |event|
      p [:close, event.code, event.reason]
      socket = nil
    end
    
    [-1, {}, []]
  else
    [200, {'Content-Type' => 'text/plain'}, ['Hello']]
  end
end

EM.run {
  thin = Rack::Handler.get('thin')
  thin.run(app, :Port => port)
}
